sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -174,17 +174,19 @@ class Dataset[T] private[sql](
     this(sqlContext.sparkSession, logicalPlan, encoder)
   }
 
-  @transient private[sql] val logicalPlan: LogicalPlan = {
+  @transient private val logicalPlan: LogicalPlan = {
```

Comment:
but they are different for commands. Now I think it's safer to create a new field in `Dataset` to add the barrier, sorry for the trouble...